libraries/FS/src/vfs_api.cpp

```diff
@@ -476,6 +476,36 @@ FileImplPtr VFSFileImpl::openNextFile(const char* mode)
     return std::make_shared<VFSFileImpl>(_fs, name.c_str(), mode);
 }
 
+boolean VFSFileImpl::seekDir(long position){
+    if(!_d){
+        return false;
+    }
+    seekdir(_d, position);
+    return true;
+}
+
+
+String VFSFileImpl::getNextFileName()
+{
+    if (!_isDirectory || !_d) {
+        return "";
+    }
+    struct dirent *file = readdir(_d);
+    if (file == NULL) {
+        return "";
+    }
+    if (file->d_type != DT_REG && file->d_type != DT_DIR) {
+        return "";
+    }
```

Comment:
This behaviour is inconsistent with `openNextFile()` which skips to the next file.

It's not possible to tell the difference between "end of file list" and "something that is not a file or directory".